### ISSUE TITLE
sudo qubesctl state.sls qvm.anon-whonix

### ISSUE THREAD

**Issue by GAhlekzis** [QubesOS/qubes-issues#4197]
### Qubes OS version:

4

### Affected component(s): 

whonix templates

---

### Steps to reproduce the behavior:

* remove preinstalled (during installation) whonix templates
* try to reinstall by issuing `sudo qubesctl state.sls qvm.anon-whonix`

### Expected behavior:

Salt states all go through green and whonix ws and gw template are installed.
Functioning sys-whonix is installed. 
Functioning anon-whonix appvm is installed.

### Actual behavior:

First state (the installation of the templates) fails because "--disablerepo=*" is not recognized. Log appended at the end.

### General notes:

I was trying to upgrade and something went wrong so i decided to start fresh. I use salt fairly often so i was quite confident this would work.
I'm really confused because no one seems to be having this issue.

Since zsh seems to be throwing the error, could it be that I need to change my shell for sys-firewall?

Sidenote: I use bash in dom0 and zsh in all my domUs.

### Log:

```
+ sudo qubesctl state.sls qvm.anon-whonix | tee anon-whonix-14-salt.log

[ERROR   ] Command '['systemd-run', '--scope', 'qubes-dom0-update', '-y', '--best', '--allowerasing', '--disablerepo=*', '--enablerepo=qubes-templates-community-testing', '--clean', '--action=install', 'qubes-template-whonix-ws-14']' failed with return code: 1
[ERROR   ] stdout: Running scope as unit: run-r50e79128dffd4dc0824105b1442e480a.scope
Using sys-firewall as UpdateVM to download updates for Dom0; this may take some time...
zsh:1: no matches found: --disablerepo=*
[ERROR   ] retcode: 1
[ERROR   ] Error occurred installing package(s). Additional info follows:

errors:
    - Running scope as unit: run-r50e79128dffd4dc0824105b1442e480a.scope
      Using sys-firewall as UpdateVM to download updates for Dom0; this may take some time...
      zsh:1: no matches found: --disablerepo=*
[WARNING ] /var/cache/salt/minion/extmods/states/ext_state_qvm.py:142: DeprecationWarning: BaseException.message has been deprecated as of Python 2.6
  status = Status(retcode=1, result=False, stderr=err.message + '\n')

[ERROR   ] ====== ['features'] ======
Virtual Machine does not exist!

====== ['tags'] ======
[SKIP] Skipping due to previous failure!
[ERROR   ] Command '['systemd-run', '--scope', 'qubes-dom0-update', '-y', '--best', '--allowerasing', '--disablerepo=*', '--enablerepo=qubes-templates-community-testing', '--action=install', 'qubes-template-whonix-gw-14']' failed with return code: 1
[ERROR   ] stdout: Running scope as unit: run-r85b1108807fa41a18c182b5c104b6312.scope
Using sys-firewall as UpdateVM to download updates for Dom0; this may take some time...
zsh:1: no matches found: --disablerepo=*
[ERROR   ] retcode: 1
[ERROR   ] Error occurred installing package(s). Additional info follows:

errors:
    - Running scope as unit: run-r85b1108807fa41a18c182b5c104b6312.scope
      Using sys-firewall as UpdateVM to download updates for Dom0; this may take some time...
      zsh:1: no matches found: --disablerepo=*
[ERROR   ] ====== ['features'] ======
Virtual Machine does not exist!

====== ['tags'] ======
[SKIP] Skipping due to previous failure!
local:
----------
          ID: template-whonix-ws-14
    Function: pkg.installed
        Name: qubes-template-whonix-ws-14
      Result: False
     Comment: Error occurred installing package(s). Additional info follows:

              errors:
                  - Running scope as unit: run-r50e79128dffd4dc0824105b1442e480a.scope
                    Using sys-firewall as UpdateVM to download updates for Dom0; this may take some time...
                    zsh:1: no matches found: --disablerepo=*
     Started: 15:03:03.564815
    Duration: 8313.015 ms
     Changes:
----------
          ID: whonix-ws-tag
    Function: qvm.vm
        Name: whonix-ws-14
      Result: False
     Comment: ====== ['features'] ======
              Virtual Machine does not exist!

              ====== ['tags'] ======
              [SKIP] Skipping due to previous failure!
     Started: 15:03:11.881016
    Duration: 29.645 ms
     Changes:
----------
          ID: whonix-ws-update-policy
    Function: file.prepend
        Name: /etc/qubes-rpc/policy/qubes.UpdatesProxy
      Result: True
     Comment: File /etc/qubes-rpc/policy/qubes.UpdatesProxy is in correct state
     Started: 15:03:11.913225
    Duration: 3.725 ms
     Changes:
----------
          ID: whonix-get-date-policy
    Function: file.prepend
        Name: /etc/qubes-rpc/policy/qubes.GetDate
      Result: True
     Comment: File /etc/qubes-rpc/policy/qubes.GetDate is in correct state
     Started: 15:03:11.917055
    Duration: 1.424 ms
     Changes:
----------
          ID: template-whonix-gw-14
    Function: pkg.installed
        Name: qubes-template-whonix-gw-14
      Result: False
     Comment: Error occurred installing package(s). Additional info follows:

              errors:
                  - Running scope as unit: run-r85b1108807fa41a18c182b5c104b6312.scope
                    Using sys-firewall as UpdateVM to download updates for Dom0; this may take some time...
                    zsh:1: no matches found: --disablerepo=*
     Started: 15:03:11.918589
    Duration: 5159.127 ms
     Changes:
----------
          ID: whonix-gw-tag
    Function: qvm.vm
        Name: whonix-gw-14
      Result: False
     Comment: ====== ['features'] ======
              Virtual Machine does not exist!

              ====== ['tags'] ======
              [SKIP] Skipping due to previous failure!
     Started: 15:03:17.077960
    Duration: 26.559 ms
     Changes:
----------
          ID: whonix-gw-update-policy
    Function: file.prepend
        Name: /etc/qubes-rpc/policy/qubes.UpdatesProxy
      Result: True
     Comment: File /etc/qubes-rpc/policy/qubes.UpdatesProxy is in correct state
     Started: 15:03:17.104646
    Duration: 1.898 ms
     Changes:
----------
          ID: sys-net
    Function: qvm.exists
      Result: True
     Comment: /usr/bin/qvm-check sys-net
              VM sys-net exists None
     Started: 15:03:17.106639
    Duration: 278.138 ms
     Changes:
----------
          ID: sys-firewall
    Function: qvm.exists
      Result: True
     Comment: /usr/bin/qvm-check sys-firewall
              VM sys-firewall exists None
     Started: 15:03:17.385269
    Duration: 286.491 ms
     Changes:
----------
          ID: sys-whonix
    Function: qvm.vm
      Result: False
     Comment: One or more requisite failed: qvm.template-whonix-gw.template-whonix-gw-14
     Changes:
----------
          ID: whonix-ws-14-dvm
    Function: qvm.vm
      Result: False
     Comment: One or more requisite failed: qvm.sys-whonix.sys-whonix, qvm.template-whonix-ws.template-whonix-ws-14
     Changes:
----------
          ID: qvm-appmenus --update whonix-ws-14-dvm
    Function: cmd.run
      Result: False
     Comment: One or more requisite failed: qvm.whonix-ws-dvm.whonix-ws-14-dvm
     Changes:
----------
          ID: anon-whonix
    Function: qvm.vm
      Result: False
     Comment: One or more requisite failed: qvm.sys-whonix.sys-whonix, qvm.template-whonix-ws.template-whonix-ws-14, qvm.whonix-ws-dvm.whonix-ws-14-dvm
     Changes:

Summary for local
------------
Succeeded: 5
Failed:    8
------------
Total states run:    13
Total run time:  14.100 s
DOM0 configuration failed, not continuing
```

**Comment by marmarek** [comment]
> zsh:1: no matches found: --disablerepo=*

This is some a difference between bash and zsh behavior. Try changing default shell back to bash in sys-firewall (or set another VM as updatevm, which do have non-zsh shell by default). If that helps, the bug is somewhere in qubes-dom0-update utility so that it uses user login shell instead of `/bin/sh` or `/bin/bash` explicitly.
Where exactly have you changed the shell? In dom0? sys-firewall's template? Have you changed /bin/sh symlink or just user shell?

**Comment by GAhlekzis** [comment]
I changed the user shell inside the fedora-28 template, because i couldn't figure out how to do it per appvm.

(I'm actually planning to revert all my changes to the preinstalled templates and working only with clones but i haven't come around to do that yet.)

I tried changing the user shell with chsh while sys-firewall is running but it doesn't seem to come into effect without a restart. So unless i change my fedora-28 template I'm basically stuck right this.
